### ISSUE TITLE
Set transform ignored after rotation_degrees has been set

### ISSUE THREAD

**Issue by goatchurchprime** [godotengine/godot#43130]
**Godot version:**
3.2.3-stable

**OS/device including version:**
Windows 10

**Issue description:**
I found a case where you can set the transform on a Spatial node to a given Transform() object, then when you print its value it is not what you set it to.  

This problem appeared as a [heisenbug](https://en.wikipedia.org/wiki/Heisenbug) because when I printed the transform just before it was set the problem went away.

Please see the code for the minimal reproduction.  It seems that if you set rotation_degrees it doesn't actually change the transform until the next read operation, and thus over-writes any direct setting transform.

**Steps to reproduce:**
See below

**Minimal reproduction project:**
```GDScript
tool
extends EditorScript

# *******
# Control-Shift X to run this code in the editor
# *******

func _run():
	var t = str2var("Transform( 0.944959, -0.327188, 0, 0, 0, 0, -0.327188, -0.944959, 0, -21.0524, -0.666294, -24.4351 )")

	var a = Spatial.new()

	# Step 1: Uncomment this line and the the transforms don't match
	#a.rotation_degrees = Vector3(-90, 0, 0)

	# Step 2: Then uncomment this line and the the transforms will match again
	#print(a.transform)
	
	print("Transforms below should match!")
	print("Transform to apply: ", t)
	a.transform = t
	print("Transform of node:  ", a.transform)
```

**Comment by akien-mga** [comment]
Confirmed that it's still reproducible in latest `3.x` (3.6-beta).
The fix proposed in #44803 might still be suitable for `3.x`.

The bug seems fixed in `master`, at least following the steps in the script gives matching transforms in all steps:
```
Transforms below should match!
Transform to apply: [X: (0.944959, 0, -0.327188), Y: (-0.327188, 0, -0.944959), Z: (0, 0, 0), O: (-21.0524, -0.666294, -24.4351)]
Transform of node:  [X: (0.944959, 0, -0.327188), Y: (-0.327188, 0, -0.944959), Z: (0, 0, 0), O: (-21.0524, -0.666294, -24.4351)]
Transforms below should match!
Transform to apply: [X: (0.944959, 0, -0.327188), Y: (-0.327188, 0, -0.944959), Z: (0, 0, 0), O: (-21.0524, -0.666294, -24.4351)]
Transform of node:  [X: (0.944959, 0, -0.327188), Y: (-0.327188, 0, -0.944959), Z: (0, 0, 0), O: (-21.0524, -0.666294, -24.4351)]
[X: (1, 0, 0), Y: (0, -0, -1), Z: (0, 1, -0), O: (0, 0, 0)]
Transforms below should match!
Transform to apply: [X: (0.944959, 0, -0.327188), Y: (-0.327188, 0, -0.944959), Z: (0, 0, 0), O: (-21.0524, -0.666294, -24.4351)]
Transform of node:  [X: (0.944959, 0, -0.327188), Y: (-0.327188, 0, -0.944959), Z: (0, 0, 0), O: (-21.0524, -0.666294, -24.4351)]
```

**Comment by akien-mga** [comment]
Fixed by #78439.
